src/fluepdot/fluepdot.py

```diff
@@ -91,15 +91,18 @@ def post_frame_raw(self, frame: str) -> Response:
 
 
     def post_frame(self, frame: List[List[bool]]) -> Response:
-        data: List[List[str]] = [[" "] * width for _ in range(height)]
-        for x, l in frame:
-            for y, b in l:
+        data: List[List[str]] = [[" "] * self.width for _ in range(self.height)]
+        for x, l in enumerate(frame):
+            for y, b in enumerate(l):
                 if b:
                     try:
-                        data[x, y] = "X"
+                        data[x][y] = "X"
                     except IndexError as e:
                         print(e)
-        return self._post(frameURL, post=data)
+        outStr = ""
```

Comment:
for loop could be inline with `outStr = "".join(["".join(line)+"\n" for line in data])`

The whole outStr could be skipped. `return self._post(frameURL, post="".join(["".join(line)+"\n" for line in data])`